internal/jobqueue/dbjobqueue/dbjobqueue.go

```diff
@@ -134,10 +142,76 @@ func New(url string) (*DBJobQueue, error) {
 		return nil, fmt.Errorf("error establishing connection: %v", err)
 	}
 
-	return &DBJobQueue{pool}, nil
+	q := &DBJobQueue{
+		pool:         pool,
+		dequeuers:    list.New(),
+		stopListener: make(chan struct{}),
+	}
+
+	go q.listen()
+
+	return q, nil
+}
+
+func (q *DBJobQueue) listen() {
+	ctx := context.Background()
```

Comment:
If this was passed in from the caller, could we not avoid having to create `cancelContext` below?

Reply:
Thanks for the suggestion, I changed the behavior to create the context in `New()`, pass it into `listen()` and store the cancel function in `DbJobQueue`. Simpler to follow with one less goroutine. :)